torchao/dtypes/nf4tensor.py

```diff
@@ -37,11 +41,248 @@ def decorator(func):
+        nf4tensor.block_size,
+        nf4tensor.n_blocks,
+        nf4tensor.scaler_block_size,
+        quantized_scalers,
+        quantization_factor,
+        scaler_mean,
+        quantized_data,
+        nf4,
+    )
+
+@implements(
+    [
+        aten.split.Tensor,
+    ]
+)
+def nf4_split(aten_op, args, kwargs=None):
+    if len(args) == 3 and args[2] != 0:
+        raise NotImplementedError(f"aten.split(NF4Tensor, dim={args[2]})")
+    nf4tensor = args[0]
+    num_chunks = nf4tensor.size(0) // args[1]
+
+    assert nf4tensor.quantized_scalers.numel() % num_chunks == 0, f"NF4Tensor.quantized_scalers.numel() not divisible by {num_chunks}"
+    quantized_scalers_chunks = aten_op(nf4tensor.quantized_scalers, nf4tensor.quantized_scalers.numel() // num_chunks, **kwargs)
+    assert nf4tensor.quantization_factor.numel() % num_chunks == 0, f"NF4Tensor.quantization_factor.numel() not divisible by {num_chunks}"
+    quantization_factor_chunks = aten_op(nf4tensor.quantization_factor, nf4tensor.quantization_factor.numel() // num_chunks, **kwargs)
+    assert nf4tensor.quantized_data.numel() % num_chunks == 0, f"NF4Tensor.quantized_data.numel() not divisible by {num_chunks}"
+    quantized_data_chunks = aten_op(nf4tensor.quantized_data, nf4tensor.quantized_data.numel() // num_chunks, **kwargs)
+
+    orig_dim = nf4tensor.dim()
+
+    if orig_dim == 1:
+        chunked_size = (nf4tensor.size(0) // num_chunks, )
+    elif orig_dim == 2:
+        chunked_size = (nf4tensor.size(0) // num_chunks, nf4tensor.size(1))
+    else:
+        chunked_size = ()
+        raise NotImplementedError(f"aten.split(NF4Tensor) wherer NF4Tensor.dim() = {orig_dim}")
+
+    return [
+        NF4Tensor(
+            SubclassTensorArgs(
+                chunked_size,
+                nf4tensor.stride(),
+                nf4tensor.storage_offset(),
+                nf4tensor.dtype,
+                nf4tensor.device,
+                nf4tensor.requires_grad,
+            ),
+            nf4tensor.block_size,
+            nf4tensor.n_blocks,
+            nf4tensor.scaler_block_size,
+            quantized_scalers,
+            quantization_factor,
+            nf4tensor.scaler_mean,
+            quantized_data,
+            nf4tensor.nf4,
+        ) for quantized_scalers, quantization_factor, quantized_data in zip(
+            quantized_scalers_chunks, quantization_factor_chunks, quantized_data_chunks
+        )
+    ]
+
+@implements(
+    [
+        aten.new_zeros.default,
+    ]
+)
+def nf4_new_zeros(aten_op, args, kwargs=None):
+    nf4_tensor = args[0]
+    new_size = args[1]
+    new_size_dim = len(new_size)
+    if (not new_size_dim in [1, 2]) or nf4_tensor.numel() % math.prod(new_size) != 0:
+        raise NotImplementedError(f"aten.new_zeros(NF4Tensor) with new size {new_size}")
+    ratio = nf4_tensor.numel() // math.prod(new_size)
+
+    assert nf4_tensor.quantized_scalers.size(0) % ratio == 0, f"quantized_scalers.numel() must be divisible by {ratio}"
```

Comment:
nit: These assertion messages preferably should include the values (i.e. both `nf4_tensor.quantized_scalers.size(0)` and `ratio`) so that they can be more actionable.